meson.build

```diff
@@ -0,0 +1,81 @@
+project('argp-standalone', 'c')
+
+argp_source = files([
+	'argp-ba.c',
+	'argp-eexst.c',
+	'argp-fmtstream.c',
+	'argp-help.c',
+	'argp-parse.c',
+	'argp-pv.c',
+	'argp-pvh.c',
+	#'argp-test.c'
+])
+
+install_headers('argp.h')
+
+conf_data = configuration_data()
+
+cc = meson.get_compiler('c')
+
+conf_data.set10('HAVE_LIMITS_H', cc.check_header('limits.h'))
+conf_data.set10('HAVE_MALLOC_H', cc.check_header('malloc.h'))
+conf_data.set10('HAVE_UNISTD_H', cc.check_header('unistd.h'))
+conf_data.set10('HAVE_CONFIG_H', true)
+conf_data.set10('HAVE_LIBINTL_H', cc.check_header('libintl.h'))
+conf_data.set10('HAVE_ALLOCA_H', cc.check_header('alloca.h'))
+# conf_data.set10('_LIBC', true)
+
+conf_data.set10('HAVE_DECL_FLOCKFILE', cc.has_function('flockfile'))
+conf_data.set10('HAVE_DECL_FPUTS_UNLOCKED', cc.has_function('fputs_unlocked', prefix : '#include <stdio.h>'))
+conf_data.set10('HAVE_DECL_FPUTC_UNLOCKED', cc.has_function('fputc_unlocked', prefix : '#include <stdio.h>'))
+conf_data.set10('HAVE_DECL_FWRITE_UNLOCKED', cc.has_function('fwrite_unlocked', prefix : '#include <stdio.h>'))
+conf_data.set10('HAVE_DECL_PUTC_UNLOCKED', cc.has_function('putc_unlocked', prefix : '#include <stdio.h>'))
```

Comment:
In this case it's reasonable to change the code to use the locking versions of the functions, IMO.

I assume you're adding these because windows doesn't support them?

Reply:
yes, but there is also code to do exactly this - fallback to the non-locked version when this variable is negative.